apps/web/src/app/domain/components/components/dynamic-anchor/dynamic-anchor.component.html

```diff
@@ -1,16 +1,21 @@
-<div class="anchor">
-  <ul>
-    <li [class.active]="activeAnchor() === 'overview'">
-      <span (click)="scrollToAnchor('overview')">Overview</span>
-    </li>
+<aside class="hidden xl:block fixed top-24 z-30 w-full shrink-0 border-l md:sticky">
+  <nav class="pl-4">
+    <ul class="grid gap-1">
+      <li class="mb-4">
+        <b class="font-medium text-primary text-lg">On this page</b>
+      </li>
+      <li [class]="'cursor-pointer transition-colors hover:text-foreground text-muted-foreground text-sm ' + (activeAnchor() === 'overview' ? 'text-primary !text-base' : '')">
+        <a (click)="scrollToAnchor('overview')">Overview</a>
+      </li>
 
-    @for (item of componentData()?.examples; track $index) {
-      <li [class.active]="activeAnchor() === item.name">
-        <span (click)="scrollToAnchor(item.name)">{{ item.name | titlecase }}</span>
+      @for (item of componentData()?.examples; track $index) {
+        <li [class]="'cursor-pointer transition-colors hover:text-foreground text-muted-foreground text-sm ' + (activeAnchor() === item.name ? 'text-primary !text-base' : '')">
+          <a (click)="scrollToAnchor(item.name)">{{ item.name | titlecase }}</a>
+        </li>
+      }
+      <li [class]="'cursor-pointer transition-colors hover:text-foreground text-muted-foreground text-sm ' + (activeAnchor() === 'api' ? 'text-primary !text-base' : '')">
```

Comment:
Should we set these classes so as not to have it repeated like it is? Why not use `ngClass` for these conditions?